src/shipping/strategies/amazon-pay-v2/amazon-pay-v2-shipping-strategy.ts

```diff
@@ -69,17 +70,19 @@ export default class AmazonPayV2ShippingStrategy implements ShippingStrategy {
         return Promise.resolve(this._store.getState());
     }
 
-    private _bindEditButton(id: string, sessionId: string, changeAction: AmazonPayV2ChangeActionType): void {
+    private _bindEditButton(id: string, sessionId: string, changeAction: AmazonPayV2ChangeActionType, shouldShowLoadingSpinner: boolean ): void {
         const button = document.getElementById(id);
 
         if (!button || !button.parentNode) {
             return;
         }
 
-        const clone = button.cloneNode(true);
-        button.parentNode.replaceChild(clone, button);
+        if (shouldShowLoadingSpinner) {
+            const clone = button.cloneNode(true);
```

Comment:
Also applies to similar code in `src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-strategy.ts`

Reply:
This was defined since the creation of the amazonpayv2 strategy: https://github.com/bigcommerce/checkout-sdk-js/pull/866#discussion_r430956512

We need to clear the original button to attach the new behavior we want when a shopper logins to amazonpay, which is getting the behavior of the amazonpay button.

Reply:
What existing behaviour are we wanting to clear?

The amazon docs don't mention cloning and infer that using `bindChangeAction` is enough to update the button:

https://developer.amazon.com/docs/amazon-pay-checkout/amazon-pay-script.html

Given it predates your change we can leave it, I'm just not sure if/why we need it.

Reply:
@leeBigCommerce As `cloneNode` does not copy event listeners, I think it's used as a way to wipe out any preexisting `onclick` behavior of the passed element and just have the one bound by the Amazon script.